### PR TITLE
Revert removal of RoleArn field in replication.Options

### PR DESCRIPTION
**Pull request by poornas**
This is needed field if `mc replicate` used for aws replication config

### CODE REVIEW

File: pkg/replication/replication.go [hunk 1]
```diff
@@ -102,6 +103,18 @@ func (c *Config) AddRule(opts Options) error {
...
 	}
+	if opts.RoleArn != "" {
+		tokens := strings.Split(opts.RoleArn, ":")
+		if len(tokens) != 6 {
+			return fmt.Errorf("invalid format for replication role Arn")
+		}
+		if strings.HasPrefix(opts.RoleArn, "arn:aws:iam") {
+			c.Role = opts.RoleArn
+		} else {
+			return fmt.Errorf("RoleArn invalid for now AWS replication configuration")
```
Comment (harshavardhana) [inline pkg/replication/replication.go:114]:
```suggestion
			return fmt.Errorf("RoleArn invalid for AWS replication configuration")
```

File: pkg/replication/replication.go [hunk 2]
```diff
@@ -102,6 +103,18 @@ func (c *Config) AddRule(opts Options) error {
 	if err != nil {
 		return err
 	}
+	if opts.RoleArn != "" {
+		tokens := strings.Split(opts.RoleArn, ":")
+		if len(tokens) != 6 {
+			return fmt.Errorf("invalid format for replication role Arn")
```
Comment (harshavardhana) [inline pkg/replication/replication.go:109]:
```suggestion
			return fmt.Errorf("invalid format for replication Rple Arn: %v", opts.RoleArn)
```

File: pkg/replication/replication.go [hunk 3]
```diff
@@ -102,6 +103,18 @@ func (c *Config) AddRule(opts Options) error {
...
+		}
+		if strings.HasPrefix(opts.RoleArn, "arn:aws:iam") {
```
Comment (harshavardhana) [inline pkg/replication/replication.go:111]: write it as `if !strings.HasPrefix(opts.RoleArn, "arn:aws:iam") { return err }`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

File: pkg/replication/replication.go [hunk 4]
```diff
@@ -102,6 +103,17 @@ func (c *Config) AddRule(opts Options) error {
 	if err != nil {
 		return err
 	}
+	if opts.RoleArn != "" {
+		tokens := strings.Split(opts.RoleArn, ":")
+		if len(tokens) != 6 {
+			return fmt.Errorf("invalid format for replication Role Arn: %v", opts.RoleArn)
+		}
+		if !strings.HasPrefix(opts.RoleArn, "arn:aws:iam") {
+			return fmt.Errorf("RoleArn invalid for AWS replication configuration")
```
Comment (harshavardhana) [inline pkg/replication/replication.go:112]:
```suggestion
			return fmt.Errorf("RoleArn invalid for AWS replication configuration: %v", opts.RoleArn)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**